### DISCUSSION TITLE
Is it possible to combine TTS and custom commands?

### DISCUSSION THREAD

**Discussion by andarotajo** [alandtse/alexa_media_player#2296] [Q&A]
So I have a little automation that has my Amazon Echo play a routine that basically just give me an overview for the weather and my calendar. The calendar was connected using Google but recently I've switched to using my new Synology for the calendar that I share via CalDav.

I've not found a way to get the calendar to my Echo device so I'm looking for ideas to replicate my routine. Before I jump into the rabbit hole that is AI I thought about using TTS for the calendar and the custom command for weather. But that doesn't seem to be possible from the documentation.

Is there a way to accomplish this, or do you perhaps have other ideas?

**Comment by digiltd** [comment]
Can you share the CalDev with your Google Calendar so that Alexa still has access to it even though you don't use Google to maintain it anymore?

**Reply by andarotajo** [reply]
Mhh not sure how. I can add the CalDav calendar on the mobile Google Calendar app but it doesn't show up when I open it on my desktop or in the Alexa app. Unfortunately I don't have external setup for my Synology Calendar set up and just for that I don't want to dive into that.

If that's the only way I'll probably need to go the TTS route

**Reply by digiltd** [reply]
>Unfortunately I don't have external setup for my Synology Calendar set up and just for that I don't want to dive into that.

Does this mean you can only access your calendar within your local network?

**Reply by andarotajo** [reply]
Kinda :D On mobile I'm using DAVx5 to sync the calendar from Synology to my phone. But since I haven't set up the calendar for remote access this only works on the local network.

So I have the calendar on my phone but the Alexa app doesn't allow me to specify a local version, only online services like Google. 

I have added the synced Synology calendar on the Google Calendar app but apparently this only stays local so it still doesn't appear in Alexa

**Reply by digiltd** [reply]
>I have added the synced Synology calendar on the Google Calendar app but apparently this only stays local so it still doesn't appear in Alexa

yeah that seems odd, i was doing some fiddling with adding a generic caldav url to my google account and seeing if it would appear in Alexa. Initially not but when i added the calendar account again I could see the newly added external url calendar (didn't remove the old one, just added a calendar expecting it to say "that account is already linked" but it added it as normal) 

if you add something to the Synology calendar using the Google Calendar app, does it also appear in whatever you usually use to view it?(sorry, not a Synology user)

I think getting it to sync with google and then having Alexa see and manage it will be the easiest route by far. Plus you will get all the extra benefits like asking "when is my next haircut appointment?".

**Reply by andarotajo** [reply]
>if you add something to the Synology calendar using the Google Calendar app, does it also appear in whatever you usually use to view it?(sorry, not a Synology user)

Yea just tried it. The main problem is that the Google Calendar app apparently doesn't sync locally added calendars to the cloud and I can't get this to work... I just want to avoid having to expose my Synology to the internet but it's hard 😄
